modules/matterhorn-kernel/src/main/java/org/opencastproject/kernel/security/LtiLaunchAuthenticationHandler.java

```diff
@@ -35,109 +37,139 @@
  * resource is received.
  */
-public class LtiLaunchAuthenticationHandler
-        implements org.springframework.security.oauth.provider.OAuthAuthenticationHandler {
+public class LtiLaunchAuthenticationHandler implements OAuthAuthenticationHandler, ManagedService {
 
   /** The logger */
   private static final Logger logger = LoggerFactory.getLogger(LtiLaunchAuthenticationHandler.class);
 
   /** The Http request parameter, sent by the LTI consumer, containing the user ID. */
-  public static final String LTI_USER_ID_PARAM = "user_id";
+  private static final String LTI_USER_ID_PARAM = "user_id";
 
-  /** The http request paramater containing the Consumer GUI **/
-  public static final String LTI_CONSUMER_GUID = "tool_consumer_instance_guid";
+  /** The http request parameter containing the Consumer GUI **/
+  private static final String LTI_CONSUMER_GUID = "tool_consumer_instance_guid";
 
-  /** LTI field containing a comma delimeted list of roles */
-  public static final String ROLES = "roles";
+  /** LTI field containing a comma delimited list of roles */
+  private static final String ROLES = "roles";
 
   /** The LTI field containing the context_id */
-  public static final String CONTEXT_ID = "context_id";
+  private static final String CONTEXT_ID = "context_id";
 
   /** The prefix for LTI user ids */
-  public static final String LTI_USER_ID_PREFIX = "lti";
+  private static final String LTI_USER_ID_PREFIX = "lti";
 
   /** The delimiter to use in generated OAUTH id's **/
-  public static final String LTI_ID_DELIMITER = ":";
+  private static final String LTI_ID_DELIMITER = ":";
 
-  /** The Matterhorn Role for OAUTH users **/
+  /** The Opencast Role for OAUTH users **/
   private static final String ROLE_OAUTH_USER = "ROLE_OAUTH_USER";
 
-  /** The default context for LTI x **/
+  /** The default context for LTI **/
   private static final String DEFAULT_CONTEXT = "LTI";
 
   /** The default learner for LTI **/
   private static final String DEFAULT_LEARNER = "USER";
 
+  /** The key to look up the admin username **/
+  private static final String ADMIN_USER_KEY = "org.opencastproject.security.admin.user";
+
+  /** The key to look up the digest username **/
+  private static final String DIGEST_USER_KEY = "org.opencastproject.security.digest.user";
+
+  /** The prefix of the key to look up a consumer key. */
+  private static final String HIGHLY_TRUSTED_CONSUMER_KEY_PREFIX = "lti.oauth.highly_trusted_consumer_key.";
+
+  /** The prefix of the key to look up a blacklisted user. */
+  private static final String BLACKLIST_USER_PREFIX = "lti.blacklist.user.";
+
+  /** The key to look up whether the admin user should be able to authenticate via LTI **/
+  private static final String ALLOW_SYSTEM_ADMINISTRATOR_KEY = "lti.allow_system_administrator";
+
+  /** The key to look up whether the digest user should be able to authenticate via LTI **/
+  private static final String ALLOW_DIGIST_USER_KEY = "lti.allow_digest_user";
```

Comment:
I noticed we have a typo in here (DIGIST - DIGEST). Would that be a problem, or was it intentionally left so?

Reply:
It's likely not intentionally but just a typo.
It should not be a problem since it's just the name of an internal constant.

Nevertheless, it would be nice if it's being fixed.
Can you submit a pull request for this against `develop`?